pagopa/io-app
# fix: [IOAPPCOM-14] Fix regression on Push Notification handling on iOS

## Short description
This PR fixes a regression on Push Notification on iOS, which tap event was not handled properly when the application is in foreground/background.

## List of changes proposed in this pull request
- Restored `react-native-push-notifications-ios` library's code on both `AppDelegate.h` and `AppDelegate.mm` [react-native-push-notifications-ios](https://github.com/react-native-push-notification/ios)

## How to test

- Start the application on the iOS simulator, make sure to be on the message list and drop an `.apns` file on the simulator with the app in the foreground. A notification popup should appear. Tapping the popup should refresh the message list (no reload spinner is shown - this is a known bug that will be fixed on another PR).

- Now put the application into background (do not kill the application). Drop the same `.apns` file on the simulator. A notification popup should appear and, upon tapping on it, the application should go into foreground and it should automatically try to load a message details (the message is identified by the `id` in the `.apns` file.

- Lastly, repeat the previous test with a killed application (expected result should be the same as before).

`.apns` example file
```
{
  "aps": {
    "alert": "Hai ricevuto un nuovo messaggio da IO",
    "badge": 0
  },
  "message_id": "0000000000001",
  "Simulator Target Bundle": "it.pagopa.app.io"
}
```
